src/js/animator/core/AutoTake.js

```diff
@@ -0,0 +1,150 @@
+   * can be detected, so the high-quality image preview can be captured off the video stream.
+   */
+  class AutoTake {
+
+    /**
+     * Initializes the AutoTake sidebar menu, and class variables.
+     */
+    static initialise() {
+      AutoTake.autoTakeDaemon = null;  // Timer Interval for detecting brightness change
+      AutoTake.enabled = true; // Is AutoTake is enabled?
+      AutoTake.isDark = false; // Is the current video frame dark (brightness below threshold)?
+      AutoTake.brightnessThreshold = 1.2;  // Brightness Threshold
+
+      // Add a list item to settings dialog
+      let autoTakeItem = document.createElement("li");
+      autoTakeList.appendChild(autoTakeItem);
+
+      // Item title
+      let itemTitle = document.createElement("div");
+      let itemTitleText = document.createElement("div");
+
+      let itemToggleBtn = document.createElement("div"); // Item toggle button
+
+      let itemSettingsContainer = document.createElement("div");
+      let brightnessThresholdInput = document.createElement("input"); // Brightness Threshold input
+      let brightnessThresholdText = document.createElement("div");
+
+      // Set title
+      autoTakeItem.appendChild(itemTitle);
+      itemTitle.classList.add("flex");
+      itemTitle.appendChild(itemTitleText);
+      itemTitleText.innerText = "Auto-Take";
+
+      // Add toggle button
+      itemTitle.appendChild(itemToggleBtn);
+      itemToggleBtn.setAttribute("style", "text-align: right");
+      itemToggleBtn.classList.add("auto-take-toggle-btn");
+      new ToggleButton(itemToggleBtn, function() {
+        if (AutoTake.enabled) {
+          clearInterval(AutoTake.autoTakeDaemon);
+          AutoTake.autoTakeDaemon = null;
+          AutoTake.enabled = false;
+        }
+        else {
+          // Start daemon to check webcam brightness 30 times per second. TODO: Make the interval time configurable
+          AutoTake.autoTakeDaemon = setInterval(AutoTake.checkBrightnessThreshold, 1000 / 30); // 30 FPS
+          AutoTake.enabled = true;
+        }
+        itemSettingsContainer.classList.toggle("hidden");
+      });
+
+      // Create item settings container to hold configurable inputs
+      autoTakeItem.appendChild(itemSettingsContainer);
+      itemSettingsContainer.classList.add("flex");
+      itemSettingsContainer.setAttribute("style", "align-items: center");
+
+      // Create brightness threshold text label and input
+      itemSettingsContainer.appendChild(brightnessThresholdText);
+      brightnessThresholdText.classList.add("flex");
+      brightnessThresholdText.setAttribute("style", "flex-basis: 80%");
+      brightnessThresholdText.innerText = "Brightness Threshold:";
+      itemSettingsContainer.appendChild(brightnessThresholdInput);
+      brightnessThresholdInput.style.margin = "0 0 0 0.5em";
+      brightnessThresholdInput.setAttribute("type", "number");
```

Comment:
Would a slider work better than a number input for this? (open to discussion)

Reply:
I tried this idea out, but found that a slider from 0-255 in the small sidebar doesn't offer enough precision for finely adjusting the brightness threshold. I debated changed the range to something like 0-10, but ultimately didn't want to restrict the users to an arbitrary range.

Reply:
I'm happy to leave it as a input then. Eventually I'm planning to implement a slider/number input combo element which could work well for this

![image](https://user-images.githubusercontent.com/8095888/114013374-5dec1480-985f-11eb-8035-b0076fc9e260.png)